R/sugar.R

```diff
@@ -20,3 +20,167 @@ tnr = function(.key, ...) {
 tnrs = function(.keys, ...) {
   dictionary_sugar_mget(mlr_tuners, .keys, ...)
 }
+
+#' @title Syntactic Sugar for Automatic Tuning
+#' 
+#' @description
+#' Function to create an [AutoTuner] object.
+#' 
+#' @param method (`character(1)`)\cr
+#'  Key to retrieve tuner from [mlr_tuners] dictionary.
+#' @param term_evals (`integer(1)`)\cr
+#'  Number of allowed evaluations.
+#' @param term_time (`integer(1)`)\cr
+#'  Maximum allowed time in seconds.
+#' @param ... (named `list()`)\cr
+#'  Named arguments to be set as parameters of the tuner.
+#' 
+#' @return [AutoTuner]
+#' 
+#' @template param_learner
+#' @template param_resampling
+#' @template param_measure
+#' @template param_search_space
+#' 
+#' @export
+#' @examples
+#' learner = lrn("classif.rpart")
+#' learner$param_set$values$minsplit = to_tune(1, 10)
+#'
+#' at = auto_tuner(
+#'   method = "random_search",
+#'   learner = learner, 
+#'   resampling = rsmp ("holdout"),
+#'   measure = msr("classif.ce"), 
+#'   term_evals = 50, 
+#'   batch_size = 10)  
+#'
+#' at$train(tsk("pima"))
+auto_tuner = function(method, learner, resampling, measure, term_evals = NULL, term_time = NULL, search_space = NULL,
+  ...) {
+  assert_choice(method, mlr_tuners$keys())
+  tuner = tnr(method, ...)
+  terminator = terminator_selection(term_evals, term_time)
+
+  AutoTuner$new(learner, resampling, measure, terminator, tuner, search_space)
+}
+
+#' @title Function for Tuning
+#' 
+#' @description
+#' Function to tune a [mlr3::Learner].
+#' 
+#' @param method (`character(1)`)\cr
+#'  Key to retrieve tuner from [mlr_tuners] dictionary.
+#' @param term_evals (`integer(1)`)\cr
+#'  Number of allowed evaluations.
+#' @param term_time (`integer(1)`)\cr
+#'  Maximum allowed time in seconds.
+#' @param ... (named `list()`)\cr
+#'  Named arguments to be set as parameters of the tuner.
+#' 
+#' @return `list()`
+#'  
+#' @template param_task
+#' @template param_learner
+#' @template param_resampling
+#' @template param_measure
+#' @template param_search_space
+#' 
+#' @export 
+#' @examples
+#' learner = lrn("classif.rpart")
+#' learner$param_set$values$minsplit = to_tune(1, 10)
+#'
+#' parameters = tune(
+#'   method = "random_search", 
+#'   task = tsk("pima"), 
+#'   learner = learner, 
+#'   resampling = rsmp ("holdout"), 
+#'   measure = msr("classif.ce"), 
+#'   term_evals = 50, 
+#'   batch_size = 10) 
+#' 
+#' # Apply hyperparameter values to learner
+#' learner$param_set$values = parameters
+tune = function(method, task, learner, resampling, measure, term_evals = NULL, term_time = NULL, search_space = NULL,
+  ...) {
+  assert_choice(method, mlr_tuners$keys())
+  tuner = tnr(method, ...)
+  terminator = terminator_selection(term_evals, term_time)
+
+  instance = TuningInstanceSingleCrit$new(task, learner, resampling, measure, terminator, search_space)
+
+  tuner$optimize(instance)
+  instance$result_learner_param_vals
```

Comment:
```suggestion
  instance
```
I know I said we care about the result but losing the instance would be kind of sad. Especially with all the plotting functions to come ;)